tools/serverpod_cli/lib/src/analyzer/entities/entity_parser/entity_parser.dart

```diff
@@ -169,19 +165,65 @@ class EntityParser {
     );
     var scope = _parseClassFieldScope(value);
     var parentTable = _parseParentTable(value);
+    var scalarField = _parseScalarField(value, fieldName);
     var isEnum = _parseIsEnumField(value);
 
-    return SerializableEntityFieldDefinition(
-      name: fieldName,
-      scope: scope,
-      type: typeResult.type..isEnum = isEnum,
-      parentTable: parentTable,
-      documentation: fieldDocumentation,
-    );
+    return [
+      if (_isRelation(value))
+        SerializableEntityFieldDefinition(
+          name: _createScalarFieldName(fieldName),
```

Comment:
Suggestion: Use `scalarField`  variable here.

This name is already available in `scalarField`. Perhaps we should also store the value of `_isRelation(...)` since we do that same parsing multiple times?